evm/script/ConfigureWormholeNtt.s.sol

```diff
@@ -0,0 +1,146 @@
+        internal
+        returns (ChainConfig[] memory config)
+    {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/cfg/WormholeNttConfig.json");
+        string memory json = vm.readFile(path);
+        bytes memory contracts = json.parseRaw(".contracts");
+
+        // Decode the json into ChainConfig array.
+        config = abi.decode(contracts, (ChainConfig[]));
+
+        // Validate values and set the contract addresses for this chain.
+        for (uint256 i = 0; i < config.length; i++) {
+            require(config[i].chainId != 0, "Invalid chain ID");
+            require(config[i].nttManager != address(0), "Invalid NTT manager address");
+            require(
+                config[i].wormholeTransceiver != address(0), "Invalid wormhole transceiver address"
+            );
+            require(config[i].inboundLimit != 0, "Invalid inbound limit");
+
+            // Make sure we don't configure the same chain twice.
+            require(!duplicateChainIds[config[i].chainId], "Duplicate chain ID");
+            duplicateChainIds[config[i].chainId] = true;
+
+            if (config[i].chainId == params.wormholeChainId) {
+                nttManager = INttManager(config[i].nttManager);
+                wormholeTransceiver = IWormholeTransceiver(config[i].wormholeTransceiver);
+            }
+        }
+    }
+
+    function configureWormholeTransceiver(
+        ChainConfig[] memory config,
+        ConfigParams memory params
+    ) internal {
+        for (uint256 i = 0; i < config.length; i++) {
+            ChainConfig memory targetConfig = config[i];
+            if (targetConfig.chainId == params.wormholeChainId) {
+                continue;
+            } else {
+                // Set relayer.
+                if (targetConfig.isWormholeRelayingEnabled) {
+                    wormholeTransceiver.setIsWormholeRelayingEnabled(targetConfig.chainId, true);
+                    console2.log("Wormhole relaying enabled for chain", targetConfig.chainId);
+                } else if (targetConfig.isSpecialRelayingEnabled) {
+                    wormholeTransceiver.setIsSpecialRelayingEnabled(targetConfig.chainId, true);
+                    console2.log("Special relaying enabled for chain", targetConfig.chainId);
+                }
+
+                // Set peer.
+                wormholeTransceiver.setWormholePeer(
+                    targetConfig.chainId, toUniversalAddress(targetConfig.wormholeTransceiver)
+                );
+                console2.log("Wormhole peer set for chain", targetConfig.chainId);
+
+                // Set EVM chain.
+                if (targetConfig.isEvmChain) {
+                    wormholeTransceiver.setIsWormholeEvmChain(targetConfig.chainId);
+                    console2.log("EVM chain set for chain", targetConfig.chainId);
+                } else {
+                   revert("Non-EVM chain is not supported yet");
```

Comment:
What about Solana?

Reply:
Happy to add Solana before we merge this.